src/main.rs

```diff
@@ -830,3 +758,133 @@ async fn build_sigstore_trust_root(
         Ok(Some(Arc::new(manual_root)))
     }
 }
+
+/*
+ * Check if the policy is already present in the local store, and if not, pull it from the remote server.
+ */
+async fn pull_if_needed(uri_or_sha_prefix: &String, matches: &ArgMatches) -> Result<()> {
+    let uri = match crate::utils::map_path_to_uri(uri_or_sha_prefix) {
+        Ok(uri) => uri,
+        Err(LookupError::PolicyMissing(_)) => match Reference::from_str(uri_or_sha_prefix) {
+            Ok(oci_reference) => {
+                info!(
+                    "cannot find policy with uri: {}, trying to pull it from remote registry",
+                    uri_or_sha_prefix
+                );
+                pull_command(uri_or_sha_prefix, matches).await?;
+                format!("registry://{}", oci_reference.whole())
+            }
+            Err(_) => {
+                return Err(anyhow!(
+                    "{}",
+                    LookupError::PolicyMissing(uri_or_sha_prefix.to_string(),)
+                ))
+            }
+        },
+        Err(e) => return Err(anyhow!("{}", e)),
+    };
+
+    match crate::utils::wasm_path(&uri) {
+        Err(LookupError::PolicyMissing(_)) => {
+            info!(
+                "cannot find policy with uri: {}, trying to pull it from remote registry",
```

Comment:
I don't understand why we try to pull twice, once in line 774 and once here.

Reply:
Same here, I think we don't need it. Do we?